apps/vx-scan/backend/src/app.ts

```diff
@@ -45,6 +56,19 @@ function buildApi(
   return grout.createApi({
     getMachineConfig,
 
+    getAuthStatus: () => auth.getAuthStatus(),
+    checkPin: (input: { pin: string }) => auth.checkPin(input),
+    readCardData: <T>({ schema }: { schema: CardDataSchemaName }) =>
+      auth.readCardData<T>({ schema: CARD_DATA_SCHEMA_MAPPING[schema] }),
+    writeCardData: <T>({
+      data,
+      schema,
+    }: {
+      data: T;
+      schema: CardDataSchemaName;
+    }) =>
+      auth.writeCardData<T>({ data, schema: CARD_DATA_SCHEMA_MAPPING[schema] }),
```

Comment:
Serialization strikes again - this time for Zod schemas. While I could add serialization support for Zod schemas to Grout, e.g. through a combo of https://github.com/StefanTerdell/zod-to-json-schema and https://github.com/StefanTerdell/json-schema-to-zod, I didn't want to increase the scope of this PR too much.

For now, I figured the frontend could just specify the schema name and the backend could look up the Zod schema given that. We do lose some type safety because of this since the frontend can no longer guarantee that the data and schema match, but again, we can always revisit.

Reply:
Mm really good point. In line with what we discussed in the backend auth spec about encapsulating as much as possible in the backend

For this case, I like the idea of a specific RPC method for saving the report data to the card